ignite/handlers/checkpoint.py

```diff
@@ -238,12 +240,13 @@ def __init__(
 
         self.to_save = to_save
         self._fname_prefix = filename_prefix + "_" if len(filename_prefix) > 0 else filename_prefix
+        self.filename_prefix = filename_prefix
```

Comment:
Normally, we should use internal attribute `_fname_prefix` as defined above, instead of introducing new attribute.
However, it is a good question whether we would like to open those init attributes like `filename_prefix`, `score_function`, `score_name`, `ext` etc... 

@sdesrozis thoughts ?

Reply:
I agree, that could be interesting.

Reply:
@sdesrozis So shall I also add others into the init attributes.

Reply:
OK, let's make all user defined attributes public:
- `self._fname_prefix -> self.filename_prefix`
- `self._score_function -> self.score_function` 
- `self._score_name -> self.score_name`
- `self._n_saved -> self.n_saved`
- `self._ext -> self.ext`

Reply:
ok will that @vfdev-5 @sdesrozis